storage/namespace.go

```diff
@@ -747,6 +748,22 @@ func (n *dbNamespace) Flush(
 	multiErr := xerrors.NewMultiError()
 	shards := n.GetOwnedShards()
 	for _, shard := range shards {
+		// This is different than calling shard.IsBootstrapped() because it was determined
+		// before the start of the tick that preceded this flush, meaning it can be reliably
+		// used to determine if all of the bootstrapped blocks have been merged / drained (ticked)
+		// and are ready to be flushed.
+		shardWasBootstrappedBeforeTick, ok := shardBootstrapStates[shard.ID()]
+		if !ok {
+			// We probably don't own this shard anymore.
+			continue
+		}
+
+		if !shardWasBootstrappedBeforeTick {
```

Comment:
nit: Perhaps combine the two if conditions together?  e.g. `if !ok || !shardWasBootstrappedBeforeTick { continue }`?

Reply:
sure

Reply:
done